hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableAirdropStore.java

```diff
@@ -0,0 +1,114 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.token.impl;
+
+import static com.hedera.node.app.service.token.impl.schemas.V0500TokenSchema.AIRDROPS_KEY;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.PendingAirdropId;
+import com.hedera.hapi.node.base.PendingAirdropValue;
+import com.hedera.node.app.spi.metrics.StoreMetricsService;
+import com.hedera.node.config.data.TokensConfig;
+import com.swirlds.config.api.Configuration;
+import com.swirlds.state.spi.WritableKVState;
+import com.swirlds.state.spi.WritableStates;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * Provides write methods for modifying underlying data storage mechanisms for
+ * working with Pending Airdrops.
+ *
+ * <p>This class is not exported from the module. It is an internal implementation detail.
+ * This class is not complete, it will be extended with other methods like remove, update etc.,
+ */
+public class WritableAirdropStore extends ReadableAirdropStoreImpl {
+    /**
+     * The underlying data storage class that holds the Pending Airdrops data.
+     */
+    private final WritableKVState<PendingAirdropId, PendingAirdropValue> airdropState;
+
+    /**
+     * Create a new {@link WritableAirdropStore} instance.
+     *
+     * @param states The state to use.
+     */
+    public WritableAirdropStore(
+            @NonNull final WritableStates states,
+            @NonNull final Configuration configuration,
+            @NonNull final StoreMetricsService storeMetricsService) {
+        super(states);
+        airdropState = states.get(AIRDROPS_KEY);
+
+        final long maxCapacity = configuration.getConfigData(TokensConfig.class).maxAllowedAirdorps();
+        final var storeMetrics = storeMetricsService.get(StoreMetricsService.StoreType.AIRDROPS, maxCapacity);
+        airdropState.setMetrics(storeMetrics);
+    }
+
+    /**
+     * Persists a new {@link PendingAirdropId} with given {@link PendingAirdropValue} into the state,
+     * as well as exporting its ID to the transaction receipt. If there is existing
+     * airdrop with the same id we add the value to the existing drop.
+     *
+     * @param airdropId    - the airdropId to be persisted.
+     * @param airdropValue - the value for the given airdropId to be persisted.
+     */
+    public void put(@NonNull final PendingAirdropId airdropId, @NonNull final PendingAirdropValue airdropValue) {
+        requireNonNull(airdropId);
+        requireNonNull(airdropValue);
```

Comment:
This can be null for NonFungibleToken type right ? Is it correct to say `requireNonNull` ?

Reply:
Different behavior now that we save AccountAirdrop instead of PendingAirdropValue